src/components/OnboardingDialog/index.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react'
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import {
+  radius,
+  shadow,
+  space,
+  duration,
+  transition,
+  color,
+  device,
+} from '../../theme'
+import { useTransition } from '../../hooks'
+import { EXITED, ENTERING, ENTERED, EXITING } from '../../hooks/useTransition'
+
+const BodyWrapper = styled.div`
+  background-color: white;
+  border-radius: ${radius.lg};
+  box-shadow: ${shadow.strong};
+  padding: ${space[32]};
+`
+
+const OnboardingDialogBody = ({ children }) => (
+  <BodyWrapper>{children}</BodyWrapper>
+)
+
+OnboardingDialogBody.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+}
+
+const DialogWrapper = styled.div`
+  position: absolute;
+  top: ${space[56]};
+  left: 0;
+  right: 0;
+  z-index: 1;
+
+  @media ${device.tablet} {
+    top: ${space[48]};
+  }
+`
+
+const OnboardingDialog = ({ children, show, ...props }) => {
+  if (!show) return null
+
+  return (
+    <DialogWrapper aria-modal="true" role="dialog" tabindex="-1" {...props}>
+      {children}
+    </DialogWrapper>
+  )
+}
+
+OnboardingDialog.defaultProps = {
+  positionTop: 56,
+}
+
+OnboardingDialog.propTypes = {
+  positionTop: PropTypes.number,
+  show: PropTypes.bool.isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+}
+
+const Container = styled.div`
+  position: relative;
+`
+
+const Backdrop = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: ${color.novaAlpha};
+  opacity: ${props =>
+    props.state === ENTERING || props.state === ENTERED ? 1 : 0};
+  transition: opacity ${transition};
+`
+
+const OnboardingWrapper = ({ children, active, state }) => {
```

Comment:
I would add `...props` here as well, since this is exported and thus needs styled components support.